packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -20,14 +27,278 @@ import {
+  /**
+   * Called when focused
+   */
+  onFocus?: (event: React.MouseEvent) => void;
+  /**
+   * Called when blurred
+   */
+  onBlur?: (event: React.MouseEvent) => void;
+  /**
+   * Called when selected value has changed
+   */
+  onChange?: (
+    option: DropdownOption | DropdownOption[],
+    event: SelectEvent | React.MouseEvent | React.KeyboardEvent
+  ) => void;
+  /**
+   * Called when the dropdown's input changes.
+   */
+  onInputChange?: (event: React.KeyboardEvent) => void;
+  /**
+   * If true, search in options will be enabled
+   */
+  searchable?: boolean;
+  /**
+   * The dropdown options
+   */
+  options?: DropdownOption[];
+  /**
+   * Text to display when there are no options
+   */
+  noOptionsMessage?: (() => string) | typeof NOOP;
+  /**
+   * If set to true, the menu will open when focused
+   */
+  openMenuOnFocus?: boolean;
+  /**
+   * If set to true, the menu will open when clicked
+   */
+  openMenuOnClick?: boolean;
+  /**
+   * If set to true, clear button will be added
+   */
+  clearable?: boolean;
+  /**
+   * custom option render function
+   */
+  optionRenderer?: (option: DropdownOption) => JSX.Element;
+  /**
+   * custom value render function
+   */
+  valueRenderer?: (...args: unknown[]) => unknown;
+  ValueRenderer?: (...args: unknown[]) => unknown;
+  /**
+   * custom menu render function
+   */
+  menuRenderer?: (...args: unknown[]) => unknown;
+  /**
+   * Default placement of the Dropdown menu in relation to its control. Use "auto" to flip the menu when there isn't enough space below the control.
+   */
+  menuPlacement?: DROPDOWN_MENU_PLACEMENT;
+
+  //  enum() PropTypes.oneOf(Object.values(DROPDOWN_MENU_PLACEMENT)),
+  /**
+   * The CSS position value of the menu, when "fixed" extra layout management might be required
+   * Fixed position can be used to solve the issue of positioning Dropdown inside overflow container like Modal or Dialog
+   */
+  menuPosition?: DROPDOWN_MENU_POSITION;
+  /**
+   * If set to true, the dropdown will be in Right to Left mode
+   */
+  rtl?: boolean;
+  /**
+   * Set default selected value
+   */
+  defaultValue?: DropdownOption[];
+  /**
+   * The component's value.
+   * When passed, makes this a [controlled](https://reactjs.org/docs/forms.html#controlled-components) component.
+   */
+  value?: DropdownOption[] | DropdownOption;
+  /**
+   * Select menu size from `Dropdown.size` - Dropdown.sizes.LARGE | Dropdown.sizes.MEDIUM | Dropdown.sizes.SMALL
+   */
+  size?: BaseSizes;
+  /**
+   * If provided Dropdown will work in async mode. Can be either promise or callback
+   */
+  asyncOptions?: (inputValue: string) => Promise<DropdownOption[]>;
+  /**
+   * If set to true, fetched async options will be cached
+   */
+  cacheOptions?: boolean;
+  /**
+   * If set, `asyncOptions` will be invoked with its value on mount and the resolved results will be loaded
+   */
+  defaultOptions?: boolean | Record<string, string>[];
```

Comment:
Sounds like should be something like:
```
defaultOptions?: GroupedOptionsType<OptionTypeBase> | OptionsType<OptionTypeBase> | boolean;
```

Reply:
Everywhere where we use `DropdownOption`, we should be caution with it
We'll might need to replace to `OptionTypeBase` as a first step, and in the future make the Dropdown component accept a generic for `OptionType` that extends `OptionTypeBase`.

For now, let's leave it as is